payment/order.go

```diff
@@ -146,6 +155,91 @@ func CreateOrderItemFromMacaroon(sku string, quantity int) (*OrderItem, error) {
 	return &orderItem, nil
 }
 
+// IsStripePayable returns true if every item is payable by Stripe
+func (order Order) IsStripePayable() bool {
+	for _, item := range order.Items {
+
+		// check stripe in payment
+		if !strings.Contains(strings.Join(item.PaymentMethods, ","), StripePaymentMethod) {
+			return false
+		}
+		// TODO: make sure we have a stripe_product_id caveat
+		// TODO: if not we need to look into subscription trials:
+		/// -> https://stripe.com/docs/billing/subscriptions/trials
+	}
+	return true
+}
+
+// CreateCheckoutSessionResponse - the structure of a checkout session response
+type CreateCheckoutSessionResponse struct {
+	SessionID string `json:"checkoutSessionId"`
+}
+
+// CreateStripeCheckoutSession - Create a Stripe Checkout Session for an Order
+func (order Order) CreateStripeCheckoutSession(email, successURI, cancelURI string) (CreateCheckoutSessionResponse, error) {
+	// Create customer if not already created
+	i := customer.List(&stripe.CustomerListParams{
+		Email: stripe.String(email),
+	})
+
+	matchingCustomers := 0
+	for i.Next() {
+		matchingCustomers++
+	}
+
+	var customerID string
+	if matchingCustomers > 0 {
+		customerID = i.Customer().ID
+	} else {
+		customer, err := customer.New(&stripe.CustomerParams{
+			Email: stripe.String(email),
+		})
+		if err != nil {
+			return CreateCheckoutSessionResponse{}, fmt.Errorf("failed to create stripe customer: %w", err)
+		}
+		customerID = customer.ID
+	}
+
+	params := &stripe.CheckoutSessionParams{
+		Customer: stripe.String(customerID),
+		PaymentMethodTypes: stripe.StringSlice([]string{
+			"card",
+		}),
+		Mode:              stripe.String(string(stripe.CheckoutSessionModeSubscription)),
+		SuccessURL:        stripe.String(successURI),
```

Comment:
should we add a query parameter containing the `order_id`?

Reply:
https://github.com/brave-intl/bat-go/pull/816/files/e2da4ad81cf34ec855abf2402ea3eab28a3ae0f7#diff-9ccdf183cadd4dfe8d34acd4d7bd8bbe5a926f5607283a9f89f3f34b5a549889R211

ClientReferenceID has the order id

Reply:
I guess my thought was it might be nice for the success page to be explicitly told which order id has just completed the checkout process ( hence I think it has to be a parameter on the SuccessURL and not just the ClientReferenceID )?

Reply:
OH.  I misunderstood, I thought you were asking about stripe getting the order id for referencing..  Will add to success url.

Reply:
resolved: https://github.com/brave-intl/bat-go/pull/816/commits/d45b5ac1be64316755d59fde7a781eb8cbadd4b7